lib/config/validation.ts

```diff
@@ -132,6 +143,18 @@ export async function validateConfig(
         message: `The "${key}" object can only be configured at the top level of a config but was found inside "${parentPath}"`,
       });
     }
+    if (optionGlobals.has(key)) {
+      if (!isGlobalConfig) {
```

Comment:
```suggestion
    if (optionGlobals.has(key)) {
      if (!isGlobalConfig) {
```

Swap this logic to avoid the .has() check when validating global config